### PR TITLE
CONSOLE-4137: Migrate console internal empty state components to patternfly

### PR DESCRIPTION
**Pull request by TheRealJon**
- Break down frontend/public/components/utils/status-box.tsx into smaller files
- Use PF EmptyState instead of internal Box component
- Remove Box component
- Rename MsgBox to ConsoleEmptyState
- Use ConsoleEmptyState for LoadingBox, LoadError, EmptyBox, and AccessDenied components
- Standardize API across ConsoleEmptyState, LoadingBox, LoadError, EmptyBox, and AccessDenied
- Improve test selector API for ConsoleEmptyState and update existing e2e tests
- Move empty state related components to console shared package
- Re-export console shared components from internal package to prevent updating hundreds of existing references.
- Improve consistency of empty state component usage

### MsgBox -> ConsoleEmptyState
before
MsgBox

after
MsgBox

---

### LoadingBox
before
LoadingBox

after
LoadingBox

---

### LoadError
before
LoadError

after
LoadError

---

EmptyBox
before
EmptyBox

after
EmptyBox

### AccessDenied
before
AccessDenied

after
AccessDenied

### CODE REVIEW

**Comment by TheRealJon** [comment]
QE Approver
/assign @yapei

Docs Approver:
/assign @opayne1

PX Approver:
/assign @reestr

**Comment by yapei** [comment]
@TheRealJon could you help rebase?

**Review by opayne1 — COMMENTED** [review]
/label docs-approved

**Comment by yapei** [comment]
@TheRealJon  one question, for LoadError, why do we change the error text to black? Isn't text in red color more distinguishable? 
Screenshot 2024-09-25 at 4 56 23 PM

**Comment by TheRealJon** [comment]
@yapei The visual updates are due to removing custom styling from our empty state components, which was part of the scope of this story. These now align with the standard PatternFly empty state pattern.

**Comment by rhamilto** [comment]
> @yapei The visual updates are due to removing custom styling from our empty state components, which was part of the scope of this story. These now align with the standard PatternFly empty state pattern.

I also notice the font size has decreased from 14px to 13px.  Given 14px is the console's default font size, we want to restore this override?

File: frontend/packages/console-shared/src/components/empty-state/ConsoleEmptyState.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+  const dataTest = props['data-test'] || 'console-empty-state';
+  const variant = props.variant || EmptyStateVariant.xs;
+  const header = (title || Icon) && (
+    <EmptyStateHeader
+      data-test={`${dataTest}-title`}
+      {...(Icon ? { icon: <EmptyStateIcon icon={Icon} /> } : {})}
+      titleText={title}
+    />
+  );
+  const bodyClassName = cx({ 'pf-v5-u-m-0': !header }); // Remove top margin if there's not a header
```
Comment (rhamilto) [inline frontend/packages/console-shared/src/components/empty-state/ConsoleEmptyState.tsx:31]: This seems like a bug in PattenFly.

**Review by rhamilto — COMMENTED** [review]
Awesome work, @TheRealJon!

File: frontend/packages/console-shared/src/components/empty-state/ConsoleEmptyState.tsx (hunk 1 above)
Comment (TheRealJon) [reply frontend/packages/console-shared/src/components/empty-state/ConsoleEmptyState.tsx:31]: Agreed. There is extra space above EmptyState components that do not have a header.

**Comment by TheRealJon** [comment]
@reestr If everything looks okay to you on this, do you mind adding your `px-approved` label?

**Pull request merged**